public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -71,6 +72,24 @@ export function transformV2(
   request: DataQueryRequest<PromQuery>,
   options: { exemplarTraceIdDestinations?: ExemplarTraceIdDestination[] }
 ) {
+  // migration for dataplane field name issue
+  response.data.forEach((f: DataFrame) => {
```

Comment:
should we wrap this behavior checking if the featureFlag exists?

Reply:
I think this would be helpful to be clear why it is here, wrapping it with a feature flag check.